### PR TITLE
issue/2799 Remove redundant behaviour and improve intellisense

### PR DESCRIPTION
**Pull request by oliverfoster**
fixes https://github.com/adaptlearning/adapt_framework/issues/2957
fixes https://github.com/adaptlearning/adapt_framework/issues/2799
helps with https://github.com/adaptlearning/adapt_framework/issues/3009

This should improve vscode intellisense substantially whilst helping us better define and utilise current Adapt APIs.

### Moved
* `Adapt.register`, `Adapt.getModelName`, `Adapt.getModelClass`, `Adapt.getViewName` and `Adapt.getViewClass` functions and `Adapt.store` property in to `core/js/components`
* `Adapt.location` property to `core/js/location` with jsdocs for the available properties
* Some deprecated `Adapt.wait` behaviour into `core/js/deprecated`
* Various bits of behaviour into their relevant modules (startController execution, etc)

### Removed
* Interim CSS fix from `scriptLoader`

### Deprecated
These now have console warnings about being deprecated
* `Adapt.a11y` for `core/js/a11y`
* `Adapt.componentStore` for `core/js/components`
* `Adapt.data` for `core/js/data`
* `Adapt.device` for `core/js/device`
* `Adapt.drawer` for `core/js/drawer`
* `Adapt.location` for `core/js/location`
* `Adapt.notify` for `core/js/notify`
* `Adapt.offlineStorage` for `core/js/offlineStorage`
* `Adapt.router` for `core/js/router`
* `Adapt.startController` for `core/js/startController`
* `Adapt.scrolling` for `core/js/scrolling`
* `Adapt.store` for `core/js/components`
* `Adapt.tracking` for `core/js/tracking`
* `Adapt.navigateToElement` for `core/js/router` equivalent
* `Adapt.register` for `core/js/components` equivalent
* `Adapt.getViewName` for `core/js/components` equivalent
* `Adapt.getViewClass` for `core/js/components` equivalent
* `Adapt.getModelName` for `core/js/components` equivalent
* `Adapt.getModelClass` for `core/js/components` equivalent
* `Adapt.findById` for `core/js/data` equivalent
* `Adapt.findViewByModelId` for `core/js/data` equivalent
* `Adapt.findByTrackingPosition` for `core/js/data` equivalent

### Changed
* jsx `html()` for `dangerouslySetInnerHtml`
* Referenced all modules directly

### Fixed
* Circular dependencies in `a11y` files and folders
* Circular dependencies in `router`, `AdaptModel`, `ContentObjectModel` and `CourseModel`
* Circular dependencies in `router` and `startController`
* Linting errors

### CODE REVIEW

File: js/data.js [hunk 1]
```diff
@@ -150,39 +152,33 @@ class Data extends AdaptCollection {
...
     try {
       Adapt.trigger('app:dataReady');
     } catch (e) {
-      Adapt.log.error('Error during app:dataReady trigger', e);
+      logging.error('Error during app:dataReady trigger', e);
     }
-    await Adapt.wait.queue();
-  }
-
-  performStartController() {
```
Comment (cahirodoherty-learningpool) [inline js/data.js:182]: Is this function no longer needed? Surely this changes the behaviour on dataReady?

Comment (cahirodoherty-learningpool) [reply js/data.js:182]: Never mind, I see the new behaviour in the startController now

Comment (oliverfoster) [reply js/data.js:182]: Yea, I just moved the responsibility to where it should be. Sorry for not explaining.

File: js/data.js [hunk 2]
```diff
@@ -236,4 +265,37 @@ class Data extends AdaptCollection {
...
+  },
+  findById: {
+    get() {
+      logging.deprecated('Adapt.findById, please use data.findById directly');
+      return data.findById;
+    }
+  },
+  findViewByModelId: {
+    get() {
+      logging.deprecated('Adapt.findViewByModelId, data.findViewByModelId directly');
```
Comment (oliverfoster) [inline js/data.js:289]: Missing `please use`

File: js/adapt.js [hunk 3]
```diff
@@ -423,6 +159,160 @@ class AdaptSingleton extends LockingModel {
...
+   * @deprecated Please use router.navigateToElement instead.
+   * @param {string} selector CSS selector of the Adapt element you want to navigate to e.g. `".co-05"`
+   * @param {Object} [settings={}] The settings for the `$.scrollTo` function (See https://github.com/flesler/jquery.scrollTo#settings).
+   * @param {Object} [settings.replace=false] Set to `true` if you want to update the URL without creating an entry in the browser's history.
+   */
+  async scrollTo() {}
+
+  /**
+   * Used to register models and views with `store`
+   * @deprecated Please use store.register instead.
```
Comment (oliverfoster) [inline js/adapt.js:254]:
```suggestion
   * @deprecated Please use components.register instead.
```

File: js/adapt.js [hunk 4]
```diff
@@ -423,6 +159,160 @@ class AdaptSingleton extends LockingModel {
...
+   * @param {string|Array} name The name(s) of the model/view to be registered
+   * @param {object} object Object containing properties `model` and `view` or (legacy) an object representing the view
+   */
+  register(name, object) {}
+
+  /**
+   * Parses a view class name.
+   * @deprecated Please use store.getViewName instead.
```
Comment (oliverfoster) [inline js/adapt.js:262]:
```suggestion
   * @deprecated Please use components.getViewName instead.
```

File: js/adapt.js [hunk 5]
```diff
@@ -423,6 +159,160 @@ class AdaptSingleton extends LockingModel {
...
+   * @param {string|Backbone.Model|Backbone.View|object} nameModelViewOrData The name of the view class you want to fetch e.g. `"hotgraphic"` or its model or its json data
+   */
+  getViewName() {}
+
+  /**
+   * Fetches a view class from the store. For a usage example, see either HotGraphic or Narrative
+   * @deprecated Please use store.getViewClass instead.
```
Comment (oliverfoster) [inline js/adapt.js:269]:
```suggestion
   * @deprecated Please use components.getViewClass instead.
```

File: js/adapt.js [hunk 6]
```diff
@@ -423,6 +159,160 @@ class AdaptSingleton extends LockingModel {
...
+   * @param {string|Backbone.Model|Backbone.View|object} nameModelViewOrData The name of the view class you want to fetch e.g. `"hotgraphic"` or its model or its json data
+   * @returns {Backbone.View} Reference to the view class
+   */
+  getViewClass() {}
+
+  /**
+   * Parses a model class name.
+   * @deprecated Please use store.getModelName instead.
```
Comment (oliverfoster) [inline js/adapt.js:277]:
```suggestion
   * @deprecated Please use components.getModelName instead.
```

File: js/adapt.js [hunk 7]
```diff
@@ -423,6 +159,160 @@ class AdaptSingleton extends LockingModel {
...
+   * @param {string|Backbone.Model|object} name The name of the model you want to fetch e.g. `"hotgraphic"`, the model to process or its json data
+   */
+  getModelName() {}
+
+  /**
+   * Fetches a model class from the store. For a usage example, see either HotGraphic or Narrative
+   * @deprecated Please use store.getModelClass instead.
```
Comment (oliverfoster) [inline js/adapt.js:284]:
```suggestion
   * @deprecated Please use components.getModelClass instead.
```

File: js/components.js [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+import Adapt from 'core/js/adapt';
+
+class Components extends Backbone.Controller {
+
+  initialize() {
+    this._register = {};
+  }
+
+  /**
+   * Used to register models and views with `Adapt.store`
```
Comment (oliverfoster) [inline js/components.js:12]:
```suggestion
   * Used to register models and views
```

File: js/data.js (hunk 2 above)
Comment (oliverfoster) [inline js/data.js:289]:
```suggestion
      logging.deprecated('Adapt.findViewByModelId, please use data.findViewByModelId directly');
```

File: js/components.js [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+  },
+  componentStore: {
+    get() {
+      logging.deprecated('Adapt.componentStore, please use core/js/components directly');
+      return components._register;
+    }
+  },
+  register: {
+    get() {
+      logging.deprecated('components.register, please use components.register instead');
```
Comment (oliverfoster) [inline js/components.js:189]:
```suggestion
      logging.deprecated('Adapt.register, please use components.register instead');
```

**Review by cahirodoherty-learningpool — APPROVED** [review]
👀

**Review by lc-alexanderbenesch — APPROVED** [review]
👍

**Review by eleanor-heath — APPROVED** [review]

**Pull request merged**